ci/Jenkinsfiles/export.groovy

```diff
@@ -174,15 +176,22 @@ pipeline {
           """
           script {
             def moduleDir = 'docker/nuxeo-explorer-export-docker'
-            withCredentials([string(credentialsId: 'instance-clid', variable: 'INSTANCE_CLID')]) {
+            withCredentials([string(credentialsId: CONNECT_EXLORER_CLID, variable: 'CONNECT_EXPLORER_CLID_VALUE'),
+                             string(credentialsId: CONNECT_EXPORT_CLID, variable: 'CONNECT_EXPORT_CLID_VALUE')]) {
               // replace lines by "--"
-              def oneLineClid = sh(
+              def oneLineExplorerClid = sh(
                 returnStdout: true,
                 script: '''#!/bin/bash +x
-                  echo -e "$INSTANCE_CLID" | sed ':a;N;\$!ba;s/\\n/--/g'
+                  echo -e "$CONNECT_EXPLORER_CLID_VALUE" | sed ':a;N;\$!ba;s/\\n/--/g'
                 '''
               )
-              withEnv(["CONNECT_EXPLORER_CLID=${oneLineClid}", "CONNECT_EXPORT_CLID=${oneLineClid}"]) {
+              def oneLineExportClid = sh(
+                returnStdout: true,
+                script: '''#!/bin/bash +x
+                  echo -e "$CONNECT_EXPORT_CLID_VALUE" | sed ':a;N;\$!ba;s/\\n/--/g'
+                '''
+              )
```

Comment:
I think a function like below could do the trick:
```
void withConnectInstanceClid(initialClid, targetClid, body) {
  withCredentials([string(credentialsId: initialClid, variable: 'CLID_MULTI')]) {
    def oneLineClid = sh(
      returnStdout: true,
      script: '''#!/bin/bash +x
        echo -e "$CLID_MULTI" | sed ':a;N;\$!ba;s/\\n/--/g'
      '''
    )
    withEnv(["$targetClid=$oneLineClid"]) {
      body.call();
    }
  }
}
```
It would be called as it:
```
withConnectInstanceClid(CONNECT_EXLORER_CLID, 'CONNECT_EXPLORER_CLID') {
  withConnectInstanceClid(CONNECT_EXPORT_CLID, 'CONNECT_EXPORT_CLID') {
    sh "envsubst < ${moduleDir}/skaffold.yaml > ${moduleDir}/skaffold.yaml~gen"
  }
}
```

Not sure it will work the first time, it is more an idea than a suggestion.

Reply:
I did see that's what you went for on the arender repo. I have two variables on my end so I did no feel that it would make things clearer/simpler.
I'm mostly curious about why a simple function executing the sh script and returning the "one line clid" value will not work out of the box here.

Reply:
Indeed, this is weird :/